### PR TITLE
e2e_node: fixes after dynamic configuration removal

### PR DESCRIPTION
**Pull request by cynepco3hahue**
#### What type of PR is this?

/kind failing-test
/kind bug

#### What this PR does / why we need it:

The PR should fix a number of tests that were broken after Dynamic Kubelet configuration removal.

Fixes:
- CPU manager
- Memory manager

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
@SergeyKanzhelev Hi, can you please review, it is the follow-up after dropping the dynamic kubelet configuration.

File: test/e2e_node/e2e_node_suite_test.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func registerNodeFlags(flags *flag.FlagSet) {
 	// It is hard and unnecessary to deal with the complexity inside the test suite.
 	flags.BoolVar(&framework.TestContext.NodeConformance, "conformance", false, "If true, the test suite will not start kubelet, and fetch system log (kernel, docker, kubelet log etc.) to the report directory.")
 	flags.BoolVar(&framework.TestContext.PrepullImages, "prepull-images", true, "If true, prepull images so image pull failures do not cause test failures.")
-	flags.BoolVar(&framework.TestContext.RestartKubelet, "restart-kubelet", true, "If true, restart Kubelet unit when the process is killed.")
+	flags.BoolVar(&framework.TestContext.RestartKubelet, "restart-kubelet", false, "If true, restart Kubelet unit when the process is killed.")
```
Comment (SergeyKanzhelev) [inline test/e2e_node/e2e_node_suite_test.go:88]: why this change?

**Review by SergeyKanzhelev — COMMENTED** [review]
lgtm except I don't understand the change of default for the flag

File: test/e2e_node/e2e_node_suite_test.go (hunk 1 above)
Comment (cynepco3hahue) [reply test/e2e_node/e2e_node_suite_test.go:88]: Now that we do not have DynamicKubelet configuration, we do not need to run a restart loop.

How did it work before:
1. Test updated kubelet via DynamicKubeletConfiguration
2. DynamicKubeletConfiguration controller applied updated configuration and stopped the kubelet.
3. The restart loop monitors the kubelet health link and once it does not respond restart the kubelet.

How does it work now:
1. The test stops the kubelet.
2. The test update kubelet static configuration.
3. The test restarts kubelet.

So now it totally tests responsibility to make sure the kubelet runs after the test. It makes restarts more predictable because it is not external to the test component that can restart the kubelet in the middle of the update configuration.

**Comment by cynepco3hahue** [comment]
@SergeyKanzhelev Thanks for the review. I answered the comment and tests passed, except one that failed because of some unrelated problem
```
W1108 09:01:32.899] Nov  8 08:33:12.987: INFO: At 2021-11-08 08:30:07 +0000 UTC - event for pod-checkpoint-no-disrupt2c1eddba-b1f2-4c82-8612-6f1dc75b9169: {kubelet n1-standard-2-cos-93-16623-39-13-2746b188} Pulled: Container image "k8s.gcr.io/pause:3.6" already present on machine
W1108 09:01:32.899] Nov  8 08:33:12.987: INFO: At 2021-11-08 08:30:07 +0000 UTC - event for pod-checkpoint-no-disrupt2c1eddba-b1f2-4c82-8612-6f1dc75b9169: {kubelet n1-standard-2-cos-93-16623-39-13-2746b188} Created: Created container pause-container
W1108 09:01:32.900] Nov  8 08:33:12.987: INFO: At 2021-11-08 08:30:07 +0000 UTC - event for pod-checkpoint-no-disrupt2c1eddba-b1f2-4c82-8612-6f1dc75b9169: {kubelet n1-standard-2-cos-93-16623-39-13-2746b188} Started: Started container pause-container
W1108 09:01:32.900] Nov  8 08:33:12.987: INFO: At 2021-11-08 08:30:10 +0000 UTC - event for pod-checkpoint-no-disrupt2c1eddba-b1f2-4c82-8612-6f1dc75b9169: {kubelet n1-standard-2-cos-93-16623-39-13-2746b188} Killing: Stopping container pause-container
W1108 09:01:32.900] Nov  8 08:33:12.987: INFO: At 2021-11-08 08:30:12 +0000 UTC - event for pod-checkpoint-no-disrupt2c1eddba-b1f2-4c82-8612-6f1dc75b9169: {kubelet n1-standard-2-cos-93-16623-39-13-2746b188} FailedKillPod: error killing pod: failed to "KillContainer" for "pause-container" with KillContainerError: "rpc error: code = Unknown desc = Error response from daemon: No such container: 154239e8b3e24d77c4d71f95c0ce4f1756c01c2daf8d6bddf9a1047ad77be4d7"
```

File: test/e2e_node/e2e_node_suite_test.go (hunk 1 above)
Comment (SergeyKanzhelev) [reply test/e2e_node/e2e_node_suite_test.go:88]: I'm thinking if there were other uses like when kubelet paniced or something. I don't remember any tests explicitly stopping kubelet. But I saw supressed panics before (https://github.com/kubernetes/kubernetes/pull/101719). 

I tend to ok this change since panic is bad enough condition to make test grid red.

**Review by SergeyKanzhelev — APPROVED** [review]
/priority important-soon
/triage accepted
/lgtm
/skip

test failures seems unrelated
